homeassistant/components/modbus/sensor.py

```diff
@@ -141,7 +145,34 @@ async def async_setup_platform(
             del entry[CONF_REGISTER_TYPE]
         config = None
 
-    for entry in discovery_info[CONF_SENSORS]:
+    for entry in discovery_info.get(CONF_BIT_SENSORS, []):
```

Comment:
Please do not mix sensor with bit_sensor !

Reply:
Please see my comment https://github.com/home-assistant/core/pull/47240/files#r604390837